packages/panzer/adapters-stk/test/stk_interface_test/tPamgenMeshFactory.cpp

```diff
@@ -321,4 +325,63 @@ TEUCHOS_UNIT_TEST(tPamgenFactory, getMeshDimension)
   TEST_EQUALITY(panzer_stk::getMeshDimension("pamgen_test.gen",MPI_COMM_WORLD,false),3);
 }
 
+
+#ifdef PANZER_HAVE_PERCEPT
+// test the ability to save the percept mesh hierarchy
+TEUCHOS_UNIT_TEST(tPamgenFactory, keepPerceptData)
+{
+  using namespace Teuchos;
+  using Teuchos::RCP;
+  using Teuchos::rcp;
+
+  {
+    out << "\nCreating pamgen mesh with parent elements." << std::endl;
+    RCP<ParameterList> p = parameterList();
+    const std::string input_file_name = "pamgen_test.gen";
+    p->set("File Name",input_file_name);
+    p->set("File Type","Pamgen");
+    p->set("Levels of Uniform Refinement",1);
+    p->set("Keep Percept Data",true);
+    p->set("Keep Percept Parent Elements",true);
+
+    RCP<STK_ExodusReaderFactory> pamgenFactory = rcp(new STK_ExodusReaderFactory());
+    TEST_NOTHROW(pamgenFactory->setParameterList(p));
+
+    RCP<STK_Interface> mesh = pamgenFactory->buildMesh(MPI_COMM_WORLD);
+    RCP<percept::PerceptMesh> refinedMesh = mesh->getRefinedMesh();
+
+    // check if the Percept data exists and has elements
+    TEST_ASSERT(nonnull(refinedMesh));
+    TEST_EQUALITY(refinedMesh->get_number_elements()>0,true);
+
+    // check if the parent information is stored
+    //TEST_ASSERT(refinedMesh->m_parent_element_field_set);
+  }
+
+
+  {
+    out << "\nCreating pamgen mesh without parent elements." << std::endl;
+    RCP<ParameterList> p = parameterList();
+    const std::string input_file_name = "pamgen_test.gen";
+    p->set("File Name",input_file_name);
+    p->set("File Type","Pamgen");
+    p->set("Levels of Uniform Refinement",1);
+    p->set("Keep Percept Data",true);
+
+    RCP<STK_ExodusReaderFactory> pamgenFactory = rcp(new STK_ExodusReaderFactory());
+    TEST_NOTHROW(pamgenFactory->setParameterList(p));
+
+    RCP<STK_Interface> mesh = pamgenFactory->buildMesh(MPI_COMM_WORLD);
+    RCP<percept::PerceptMesh> refinedMesh = mesh->getRefinedMesh();
+
+    // check if the Percept data exists and has elements
+    TEST_ASSERT(nonnull(refinedMesh));
+    TEST_EQUALITY(refinedMesh->get_number_elements()>0,true);
+
+    // check if the parent information is stored 
+    //TEST_ASSERT(!refinedMesh->m_parent_element_field_set);
```

Comment:
Can you add a check to make sure the parent element data exists? It looks like the first new option, "Keep Percept Data" is tested here but the "Keep Percept Parent Elements" option is not.